benchexec/tools/ultimate.py

```diff
@@ -105,24 +108,21 @@ def _ultimate_version(self, executable):
                 "--version",
             ],
             # 1
-            [
-                self.get_java(),
-                "-Xss4m",
-                "-jar",
-                launcher_jar,
-                "-data",
-                data_dir,
-                "--version",
-            ],
+            ["-Xss4m", "-jar", launcher_jar, "-data", data_dir, "--version"],
         ]
 
         self.api = len(cmds)
         for cmd in cmds:
-            version = self._query_ultimate_version(cmd, self.api)
-            if version != "":
-                return version
+            for java_version, java in java_versions.items():
+                version = self._query_ultimate_version([java] + cmd, self.api)
+                if version != "":
```

Comment:
```suggestion
                if version:
```
This would also cover cases like `None` being returned.